server/sonar-process-monitor/src/main/java/org/sonar/application/config/JdbcSettings.java

```diff
@@ -110,7 +111,7 @@ Provider resolveProviderAndEnforceNonnullJdbcUrl(Props props) {
   }
 
   private static String buildH2JdbcUrl(int embeddedDatabasePort) {
-    return "jdbc:h2:tcp://localhost:" + embeddedDatabasePort + "/sonar";
+    return "jdbc:h2:tcp://" + InetAddress.getLoopbackAddress().getHostAddress() + ":" + embeddedDatabasePort + "/sonar";
```

Comment:
Small note here, with IPv6 is requiring the host to be separated with bracket (because : is used in IPv6 address)

cf. http://www.ietf.org/rfc/rfc3986.txt